aea/mail/base.py

```diff
@@ -334,9 +378,43 @@ def __init__(
         )
         if isinstance(message, Message):
             message = self._check_consistency(message, to, sender)
+
         self._to = to
         self._sender = sender
-        self._protocol_id = protocol_id
+
+        # set protocol_id and protocol_specification_id
+        if isinstance(message, Message):
+            # protocol_id provided as an argument in priority
+            # use Message.protocol_id only if no protocol_id provided
+            protocol_id = protocol_id or Message.protocol_id
+
```

Comment:
Somewhere enforce that message is either `bytes` or `Message` type

Reply:
done